lib/process-services-cloud/src/lib/form/components/widgets/attach-file/attach-file-cloud-widget.component.spec.ts

```diff
@@ -398,106 +237,126 @@ describe('AttachFileCloudWidgetComponent', () => {
+            clickOnAttachButton('attach-file-alfresco');
             fixture.detectChanges();
             await fixture.whenStable();
-            const attachButton: HTMLButtonElement = element.querySelector('#attach-file-alfresco');
+            fixture.detectChanges();
 
-            expect(attachButton).not.toBeNull();
+            const alias = '-root-';
+            const opt = { relativePath: '/pathBasedOnStringvariablevalue' };
+            expect(fetchNodeIdFromRelativePathSpy).toHaveBeenCalledWith(alias, opt);
+            expect(widget.rootNodeId).toEqual('mock-string-value-node-id');
+        });
 
-            attachButton.click();
+        it('should be able to use default location if mapped string variable value is undefined/empty', async () => {
+            const fetchNodeIdFromRelativePathSpy = spyOn(contentCloudNodeSelectorService, 'fetchNodeIdFromRelativePath').and.returnValue(mockNodeId);
+            createUploadWidgetField(new FormModel(), 'attach-file-alfresco', [], allSourceWithStringTypeEmptyValue);
+            fixture.detectChanges();
             await fixture.whenStable();
+
+            clickOnAttachButton('attach-file-alfresco');
             fixture.detectChanges();
+            await fixture.whenStable();
 
-            expect(widget.rootNodeId).toEqual('-root-');
-            expect(openUploadFileDialogSpy).toHaveBeenCalledWith('-root-', 'single', true, true);
+            const alias = '-my-';
+            const opt = { relativePath: '' };
+            expect(fetchNodeIdFromRelativePathSpy).toHaveBeenCalledWith(alias, opt);
+            expect(widget.rootNodeId).toEqual('mock-node-id');
         });
 
-        it('Should set default user alias (-my-) as rootNodeId if destinationFolderPath contains wrong alias and single upload for Alfresco Content + Locale', async () => {
-            widget.field = new FormFieldModel(new FormModel(), {
-                type: FormFieldTypes.UPLOAD,
-                value: []
-            });
-            widget.field.id = 'attach-file-alfresco';
-            widget.field.params = <FormFieldMetadata> allSourceWithWrongAliasParams;
-            widget.field.params.multiple = false;
+        it('should be able to use mapped folder variable value if destinationFolderPath set to folder type variable', async () => {
+            const fetchNodeIdFromRelativePathSpy = spyOn(contentCloudNodeSelectorService, 'fetchNodeIdFromRelativePath');
+            const form = new FormModel({ formVariables, processVariables});
+            createUploadWidgetField(form, 'attach-file-alfresco', [], mockAllFileSourceWithFolderVariablePathType);
             fixture.detectChanges();
             await fixture.whenStable();
-            const attachButton: HTMLButtonElement = element.querySelector('#attach-file-alfresco');
+            clickOnAttachButton('attach-file-alfresco');
+            fixture.detectChanges();
+            await fixture.whenStable();
+            fixture.detectChanges();
```

Comment:
why do you need to call again fixture.detectChanges twice?

Reply:
Updated, Thanks